concepts/rxjs-primer.md

```diff
@@ -0,0 +1,521 @@
+  // called once on completion
+  complete: () => console.log('complete!')
+});
+```
+
+It's important to note that each subscription will create a new execution
+context. This means calling `subscribe` a second time will create a new event
+listener:
+
+```js
+// addEventListener called
+const subscription = myObservable.subscribe(event => console.log(event));
+
+// addEvent listener called again!
+const secondSubscription = myObservable.subscribe(event => console.log(event));
+
+// clean up with unsubscribe
+subscription.unsubscribe();
+secondSubscription.unsubscribe();
+```
+
+By default, a subscription creates a one on one, one-sided conversation between
+the observable and observer. This is like your boss (the observable) yelling
+(emitting) at you (the observer) for merging a bad PR. This is also known as
+**unicasting**. If you prefer a conference talk scenario - one observable, many
+observers - you will take a different approach which includes **multicasting**
+with `Subjects` (either explicitly or behind the scenes). More on that in a
+future article!
+
+It's worth noting that when we discuss an Observable source emitting data to
+observers, this is a push based model. The source doesn't know or care what
+subscribers do with the data, it simply pushes it down the line.
+
+While working on a stream of events is nice, it's only so useful on it's own.
+**What makes RxJS the "lodash for events" are its...**
+
+## Operators
+
+Operators offer a way to manipulate values from a source, returning an
+observable of the transformed values. Many of the RxJS operators will look
+familiar if you are used to JavaScripts `Array` methods. For instance, if you
+want to transform emitted values from an observable source, you can use
+[`map`](../operators/transformation/map.md):
+
+```js
+// import the from operator
+import { of } from 'rxjs';
+/*
+ *  'of' allows you to deliver values in a sequence
+ *  In this case, it will emit 1,2,3,4,5 in order.
+ */
+const dataSource = of(1, 2, 3, 4, 5);
+
+// subscribe to our source observable
+const subscription = dataSource
+  .pipe(
+    // add 1 to each emitted value
+    map(value => value + 1)
+  )
+  // log: 2, 3, 4, 5, 6
+  .subscribe(value => console.log(value));
+```
+
+Or if you want to filter for specific values, you can use
+[`filter`](../operators/filtering/filter.md):
+
+```js
+// import the from operator
+import { of } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+const dataSource = from(1, 2, 3, 4, 5);
+
+// subscribe to our source observable
+const subscription = dataSource
+  .pipe(
+    // only accept values 2 or greater
+    filter(value => value >= 2)
+  )
+  // log: 2, 3, 4, 5
+  .subscribe(value => console.log(value));
+```
+
+In practice, if there is a problem you need to solve, it's more than likely
+**there is an operator for that**. And while the sheer number of operators can
+be overwhelming as you begin your RxJS journey, you can narrow it down to a
```

Comment:
maybe worthwhile referencing operator decision tree ? (https://rxjs-dev.firebaseapp.com/operator-decision-tree)

Reply:
Ah, good idea! I’ll add this in